docs/arcaflow/workflows/flow-control.md

```diff
@@ -2,6 +2,258 @@
 
 Flow control allows the workflow author to build a workflow with a decision tree based on supported flow logic. These flow control operations are not implemented by plugins, but are part of the workflow engine itself.
 
+## Conditional Step Execution
+
+Conditional step execution can be achieved with the `enabled` input.
+
+The value passed into the `enabled` field should be an expression that resolves as a boolean. So it should either reference a path to a boolean value, or it should be a binary comparison.
+
+A situation that would benefit from conditional step execution includes a step that uploads the result to a remote server. For development runs you may not want to upload the results, so you can create a `bool` field in the input to toggle the upload, then you can reference that field in the `enabled` field of the upload step.
+
+Example:
+```yaml title="workflow.yaml"
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        step_1_enabled:
+          type:
+            type_id: bool
+        step_1_input:
+          type:
+            type_id: integer
+steps:
+  my_step:
+    plugin:
+      deployment_type: image
+      src: path/to/my_plugin:1
+    input:
+      param_1: !expr $.input.step_1_input
+    enabled: !expr $.input.step_1_enabled
+outputs:
+  workflow_success:
+    plugin_output: !expr $.steps.my_step.outputs.success
+```
+
+But oh no! The workflow fails with the error `all outputs marked as unresolvable` when the step is disabled.
+Why is this? It's simple. The step did not run, as instructed, and therefore the output `workflow_success` cannot get the information it needs. See the next section for methods of handling this situation.
```

Comment:
Instead of the rhetorical "Why is this? It's simple." call/response, I think it might read better to simply set up that "This happens because the workflow instructed the step to not run..." Asking the reader a question is fine for a quiz or something, but inline like this it can easily come off as a bit condescending.